resharper/resharper-unity/src/CSharp/Daemon/Stages/BurstCodeAnalysis/BurstCodeAnalysisUtil.cs

```diff
@@ -0,0 +1,64 @@
+using JetBrains.Annotations;
+using JetBrains.ReSharper.Features.Navigation.Features.GoToDeclarationUsages.Impl;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.CSharp.Util;
+using JetBrains.ReSharper.Psi.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Stages.BurstCodeAnalysis
+{
+    public static class BurstCodeAnalysisUtil
+    {
+        public static bool IsSuitableForBurst(this IType type)
+        {
+            return type.IsValueType() || type.IsPointerType() || type.IsOpenType;
+        }
+
+        public static bool IsQualifierOpenType(IConditionalAccessExpression conditionalAccessExpression)
+        {
+            if (conditionalAccessExpression is IInvocationExpression)
+            {
+                conditionalAccessExpression =
+                    conditionalAccessExpression.ConditionalQualifier as IConditionalAccessExpression;
+            }
+
+            return conditionalAccessExpression?.ConditionalQualifier?.Type().IsOpenType ?? false;
+        }
+    
+        private static bool MayContainProhibitedMethods(this ITypeElement typeElement)
+        {
+            //                                             CGTD is valueType is what i want? what is this anyway?
+            return typeElement is IClass @class && (@class.IsSystemValueTypeClass() || @class.IsObjectClass());
```

Comment:
Looks like magic, could you add comments?

Reply:
redesigned, comments added